### PR TITLE
Fix for infinite loop bug

### PR DESCRIPTION
**Pull request by AdamFx990**
Worlds smallest pull request.

Fixes issue:  #608

### CODE REVIEW

**Comment by FakelsHub** [comment]
Wrong fix 

**Return value**
Iterator following the last removed element.
If pos refers to the last element, then the end() iterator is returned.
If last==end() prior to removal, then the updated end() iterator is returned.
If [first, last) is an empty range, then last is returned. 

https://en.cppreference.com/w/cpp/container/list/erase

**Comment by AdamFx990** [comment]
When you say to _"check [my] compiler capabilities"_: do you mean there might be a discrepancy in how this code is executed between compilers? O.o 

Are you able to replicate this issue?

**Comment by FakelsHub** [comment]
> Could you please elaborate on what you mean when you say

just check "your" compiler for a possible bug
https://rextester.com/VCV91745 <- how it should work.

**Comment by AdamFx990** [comment]
Thanks for the snippet. I compiled it with clang and g++ on my machine and the output was fine. 

Prior to 31d3351198f1dda08bca6628aac678ffca703bc1, the iterator behaves as intended.

**Comment by alexeevdv** [comment]
Hey, thanks for the contribution. I see one major issue with you code. It adds dependency on global object (Game) to timer class. You should pass this dependency into constructor or use getters/setters to change timer object state

File: src/Game/Timer.h [hunk 1]
```diff
@@ -32,6 +32,7 @@ namespace Falltergeist
                 bool _repeat = false;
                 float _interval = 0;
                 float _timeTracked = 0;
+                uint  _lastTick;
```
Comment (alexeevdv) [inline src/Game/Timer.h]: use uint32_t please

**Comment by AdamFx990** [comment]
There's been significant performance regressions on maps where _Opcode80F0_ is fired a lot.

**Comment by alexeevdv** [comment]
#608 seems to be fixed

**Pull request closed without merging**